plugins/secretstores/docker/docker.go

```diff
@@ -0,0 +1,85 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package docker
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/secretstores"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Docker struct {
+	ID   string `toml:"id"`
+	Path string `toml:"path"`
+}
+
+func (*Docker) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init initializes all internals of the secret-store
+func (d *Docker) Init() error {
+	if d.ID == "" {
+		return errors.New("id missing")
+	}
+	if d.Path == "" {
+		// setting the default directory for Docker Secrets
+		// if no explicit path mentioned in configuration
+		d.Path = "/run/secrets"
+	}
+	if _, err := os.Stat(d.Path); os.IsNotExist(err) {
+		// if there is no /run/secrets directory for default Path value
+		// this implies that there are no secrets.
+		// Or for any explicit path definitions for that matter.
+		return fmt.Errorf("directory %s does not exist", d.Path)
+	}
```

Comment:
What about other errors? Like missing permission etc?

Reply:
I have added a `os.IsPermission` check. I am not sure how to test this actually with the `testdata` directory

Reply:
Tests conducted in a live container shows the following error:

```bash
 cannot read the secret's value under the directory: open /run/secrets/opcua_password: permission denied
```